File: Task/Map-range/Euphoria/map-range.ex
function map_range(sequence a, sequence b, atom s)
    return b[1]+(s-a[1])*(b[2]-b[1])/(a[2]-a[1])
end function

for i = 0 to 10 do
    printf(1, "%2g maps to %4g\n", {i, map_range({0,10},{-1,0},i)})
end for

File: Task/Map-range/Pascal-P/map-range.pas
program map(output);
(* Map range *)
var
  i: integer;

  function maprange(a1, a2, b1, b2: real; s: real): real;
  begin
    maprange := (s - a1) * (b2 - b1) / (a2 - a1) + b1;
  end;

begin
  for i := 0 to 10 do
    writeln(i: 2, ' maps to: ', maprange(0.0, 10.0, -1.0, 0.0, i): 10);
end.
